src/tasks.py

```diff
@@ -158,24 +184,55 @@ def command(
 
     input_files_map = {}
     for input_file in input_files:
-        input_files_map[input_file.get(
-            "path", input_file.get("uuid", "UNKNOWN FILE"))] = input_file.get(
-                "display_name", "UNKNOWN FILE NAME")
+        input_files_map[
+            input_file.get("path", input_file.get("uuid", "UNKNOWN FILE"))
+        ] = input_file.get("display_name", "UNKNOWN FILE NAME")
 
-    folders_and_files = []
-    for input_file in input_files:
-        if 'path' not in input_file:
-            logger.warning("Skipping file %s as it does not have an path", input_file)
-            continue
-        folders_and_files.append('--folder')
-        folders_and_files.append(input_file.get('path'))
-
-    cmd = ['fraken'] + folders_and_files + [f'{all_yara.path}']
-    with open(fraken_output.path, 'w+', encoding="utf-8") as log:
-        process = subprocess.Popen(cmd, stdout=log)
-        process.wait()
-
-    with open(fraken_output.path, 'r', encoding="utf-8") as json_file:
+    try:
+        folders_and_files = []
+        bd = None
+        disks_mounted = []
+        for input_file in input_files:
+            if "path" not in input_file:
+                logger.warning(
+                    "Skipping file %s as it does not have an path", input_file
+                )
+                continue
+
+            input_file_path = input_file.get("path")
+            # Check if disk image, mount and add mountpoints to scan
+            if mount_disk_images and is_disk_image(input_file):
+                bd = BlockDevice(input_file_path, min_partition_size=1)
+                bd.setup()
+                mountpoints = bd.mount()
+                disks_mounted.append(bd)
+
+                if not mountpoints:
+                    logger.info(
+                        "No mountpoints returned for input file %s",
+                        input_file.get("display_name"),
+                    )
+                for mountpoint in mountpoints:
+                    folders_and_files.append("--folder")
```

Comment:
Does this mean it could pass multiple "--folder" params in to the scanner? I'm not sure I've tested that

Reply:
Note: The current logic to find a file owner (checking /etc/passwd) will probably not be that useful in case of disk mounts.

Reply:
Ok yes seems I did multiple folders into account
It should be looking for /etc/passwd under the given folders, though.